src/api/app/views/webui/distributions/new.html.haml

```diff
@@ -0,0 +1,37 @@
+- @pagetitle = "Add distribution repository to #{@project}"
+
+
+.card.mb-3
+  = render partial: 'webui/project/tabs', locals: { project: @project }
+  .card-body
+    %h3 Add Repositories to #{@project}
+    .row
+      - @distributions.each do |vendor, distributions|
+        .col-md-12
+          .pb-4
+            %h6.pb-2
+              - if resolve_asset_path("distributions/#{vendor.downcase}.svg")
+                = image_tag("distributions/#{vendor.downcase}.svg")
+              #{vendor} distributions
+            .row
+              - distributions.group_by(&:reponame).each do |_reponame, variants|
+                .col-md-3
+                  .pb-2
+                    - if variants.length > 1
+                      = variants.first.reponame
+                      Variants
+                      %i.fa.fa-question-circle.text-info{ data: { placement: 'top',
+                                                                  toggle: 'popover',
+                                                                  content: 'Variants are distributions of the same general kind that only
+                                                                            differ in the set of packages they include. You should consult
+                                                                            the distributions documentation to learn more about the variants.
+                                                                            It is only possible to add one distribution variant as repository.'
+                                                                } }
+                      \:
+                    - variants.each do |variant|
+                      = form_tag(toggle_project_distributions_path, remote: true, id: "distribution-#{variant.id}-form") do
+                        - checked = @project.has_distribution(variant.project, variant.repository) ? 'checked' : nil
+                        = render partial: 'variant', object: variant, locals: { checked: checked }
```

Comment:
if you could have the whole section from line [17](https://github.com/openSUSE/open-build-service/pull/11250/commits/da6e9a6aaeff23c9dbd741354beb719e18606b08#diff-685a2370d164f72f6d8bb4fa79af1561e894b348f1adaa120b1387a527586a78R17) upto here in a partial, then you didn't have to render a partial inside of a each loop, which normally impacts badly the performance..

Reply:
I would but I can't group a collection AFAIK. The performance impact is also bearable, the number of distributions is rather static.